File: Algo_Method/calc_bits/ch04/Q8/Q8.rb
n, s = gets.chomp.split.map(&:to_i)
q = gets.chomp.to_i
q.times do
    c, x = gets.chomp.split.map(&:to_i)
    if c == 0
        s |= (1 << x)
    elsif c == 1
        s &= ~(1 << x)
    else
        puts (s & (1 << x)) == 0 ? "off" : "on"
    end
end

File: Algo_Method/calc_bits/ch04/Q8/Q8.py
n, s = map(int, input().split())
q = int(input())
for i in range(q):
    c, x = map(int, input().split())
    if c == 0: s |= (1 << x)
    elif c == 1: s &= ~(1 << x)
    else: print("on" if (s & (1 << x)) else "off")